ktor-server/ktor-server-plugins/ktor-server-websockets/jvm/test/io/ktor/tests/websocket/WebSocketTest.kt

```diff
@@ -53,12 +55,152 @@ class WebSocketTest {
+                    }
+                }
+            }
+        }
+    }
+
+    @Test
+    fun testSerializationWithNoConverter() {
+        withTestApplication {
+            application.install(WebSockets) {}
+
+            application.routing {
+                webSocket("/echo") {
+                    assertFailsWith<WebsocketConverterNotFoundException>("No converter was found for websocket") {
+                        receiveDeserialized<Data>()
+                    }
+                    assertFailsWith<WebsocketConverterNotFoundException>("No converter was found for websocket") {
+                        sendSerialized(Data("hello", 12))
+                    }
+                    outgoing.send(Frame.Close())
+                }
+            }
+
+            val jsonData = "{\"string\":\"hello\",\"count\":123}"
+            val sendBuffer = ByteBuffer.allocate(jsonData.length + 100)
+
+            Serializer().apply {
+                enqueue(Frame.Text(jsonData))
+                serialize(sendBuffer)
+            }
+
+            val conversation = Job()
+
+            handleWebSocket("/echo") {
+                bodyChannel = writer {
+                    channel.writeFully(sendBuffer.array())
+                    channel.flush()
+                    conversation.join()
+                }.channel
+            }.let {
+                runBlocking {
+                    withTimeout(Duration.ofSeconds(10).toMillis()) {}
+                }
+            }
+        }
+    }
+
+    @Test
+    fun testDeserializationWithOnClosedChannel() {
+        withTestApplication {
+            application.install(WebSockets) {
+                contentConverter = GsonWebsocketContentConverter()
+            }
+
+            application.routing {
+                webSocket("/echo") {
+                    assertFailsWith<ClosedReceiveChannelException> {
+                        receiveDeserialized<Data>()
+                    }
+                    outgoing.send(Frame.Close())
+                }
+            }
+            val sendBuffer = ByteBuffer.allocate(100)
+
+            Serializer().apply {
+                enqueue(Frame.Text(""))
+                serialize(sendBuffer)
+            }
+
+            val conversation = Job()
+
+            handleWebSocket("/echo") {
+                bodyChannel = writer {
+                    channel.writeFully(sendBuffer.array())
+                    channel.flush()
+                    conversation.join()
+                }.channel
+            }.let {
+                runBlocking {
+                    withTimeout(Duration.ofSeconds(10).toMillis()) {}
+                }
+            }
+        }
+    }
+
     @Test
     fun testFrameSize() {
         withTestApplication {
             application.install(WebSockets)
+
             application.routing {
                 webSocketRaw("/echo") {
+                    application.plugin(WebSockets).maxFrameSize
```

Comment:
why is it here?

Reply:
There was an incorrect  merge with `main`. This line is removed now